core/variables.js

```diff
@@ -324,9 +328,44 @@ Blockly.Variables.createVariableButtonHandler = function(
   };
   promptAndCheckWithAlert('');
 };
+
+/**
+ * Handles "Delete Variable" button in the default variables toolbox category.
+ * It will prompt the user for a varibale name, including re-prompts if a name
+ * does not exist.
+ *
+ * @param {!Blockly.Workspace} workspace The workspace on which to delete the
+ *     variable.
+ */
+Blockly.Variables.deleteVariableButtonHandler = function(workspace) {
+  var promptAndCheckWithAlert = function(defaultName) {
+    Blockly.Variables.promptName(Blockly.Msg['DELETE_VARIABLE_PROMPT'], defaultName,
```

Comment:
Instead of using the existing text field prompt can you make one with a dropdown with all existing variables?